src/pytest_bdd/scenario.py

```diff
@@ -199,15 +199,19 @@ def _execute_step_function(
             f"Unexpected `NoneType` returned from " f"parse_arguments(...) in parser: {context.parser!r}"
         )
 
-        if step.datatable is not None:
-            kwargs["datatable"] = step.datatable.raw()
-
         for arg, value in parsed_args.items():
             if arg in converters:
                 value = converters[arg](value)
             kwargs[arg] = value
 
+        if step.datatable is not None:
+            kwargs["datatable"] = step.datatable.raw()
```

Comment:
maybe we should raise an error or a warning in case the `datatable` or `docstring` param clashes with some of the params from the step parser?

not for this PR though

Reply:
Agreed.  I'll pop a new PR for this